java-checks/src/main/java/org/sonar/java/checks/ServletInstanceFieldCheck.java

```diff
@@ -47,17 +53,56 @@
 @SqaleConstantRemediation("30min")
 public class ServletInstanceFieldCheck extends SubscriptionBaseVisitor {
 
+  private List<VariableTree> issuableVariables = new ArrayList<>();
+  private List<VariableTree> excludedVariables = new ArrayList<>();
+
   @Override
   public List<Kind> nodesToVisit() {
-    return ImmutableList.of(Tree.Kind.VARIABLE);
+    return ImmutableList.of(Tree.Kind.VARIABLE, Kind.METHOD);
+  }
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    super.scanFile(context);
+    reportIssuesOnVariable();
   }
 
   @Override
   public void visitNode(Tree tree) {
-    VariableTree variable = (VariableTree) tree;
-    if (hasSemantic() && isOwnedByAServlet(variable) && !isStaticOrFinal(variable)) {
+    if (tree.is(Kind.METHOD) && isServletInit((MethodTree) tree)) {
+      tree.accept(new AssignmentVisitor());
+    } else if (tree.is(Kind.VARIABLE)) {
+      VariableTree variable = (VariableTree) tree;
+      if (hasSemantic() && isOwnedByAServlet(variable) && !isStaticOrFinal(variable)) {
```

Comment:
checking for the semantic should be done before checking the type of the tree, as semantic is used to get the parameter type of the method.